src/client/live_price.ts

```diff
@@ -1,53 +1,209 @@
-import { Client } from './client';
-import { Region } from './collections';
-import { v4 as uuidv4 } from 'uuid';
-
-function getLivePrice<T>(
-  client: Client,
-  symbols: string[],
-  region: Region,
-  streamId: string = uuidv4(),
-): {
-  events: AsyncIterable<T>,
-  cancel: () => void
-} {
-  const url = `${client["baseUrl"]}/api/v1/stock/price/live?filter=${symbols.join(',')}&region=${region}&stream=${streamId}`;
-
-  return client.sendSSERequest<T>(url);
+import { Client } from "./client";
+import { Region } from "./collections";
+import WebSocket from "ws";
+
+type WebSocketMessage<T> = {
+  type: string;
+  message: T;
+};
+
+export class LivePriceWebSocketService {
```

Comment:
Bu `class`'in API'i gereksiz kisitlayici. Misal ben bu class'i kullaniyorum.

Hisse sayfasina girdim, o hisseye subscribe olmak istiyorum ki grafigi ve returnleri update edeyim.

Ama bir de asagida "Benzer Hisseler" var. Onlara da subscribe olmak istiyorum, ama grafiklerini degil sadece gunluk degisimini update etmek istiyorum.

Nasil bir `getLivePrice()` callu yapmaliyim? Su sekilde sacma bir kod yazmam gerekecek:

```ts
getLivePrice(
   [stock.symbol, ...stock.similar.map((s) => s.symbol)],
   (data) => {
       if (data.symbol === stock.symbol) {
             // handle graph 
             return
       }
       // handle similar
   }
```

Bir de soyle bir durum var, diyelim XU100'e endeksi gostermek icin her turlu subscribe olmak istiyorum, onun logicini de hatirlayip tekrardan bu fonksiyonun icine koymam lazim (ki baska component'ta muhtemelen, neredeyse imkansiz bir duruma donusuyor)

Onun yerine bu class'in API'i soyle olsa:

`type Handler (data: BISTLiveStockData) => void`

`static subscribe(symbols: string[], handler: Handler)`

ve,

`static unsubscribe(symbols: string[])`

Her `symbol` icin class'in icerisinde list of `Handler` tutulabilir.

`handlers: Record<string, Handler[]>`

Sonrasinda `ws.onmessage()`, hangi symbol olduguna gore bu handler'lar sirayla cagrilir.

Neden boyle? Cunku diyelim GARAN'in fiyatini hem grafigi update etmek icin hem baska bir seyi baska bir yerde update etmek isteyebilirim. Iki farkli component olabilir bunlar.